client/components/async-load/index.jsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { omit } from 'lodash';
+
+export default class AsyncLoad extends Component {
+	static propTypes = {
+		require: PropTypes.func.isRequired,
+		placeholder: PropTypes.node
+	};
+
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			component: null
+		};
+	}
+
+	componentWillMount() {
```

Comment:
this may also need to have `componentWillReceiveProps` in case we get into a situation where React decides to reuse an instance instead of spinning up a new one. Given how this will be used, it's probably unlikely, but worth doing.